mk/compile.mk

```diff
@@ -230,4 +230,57 @@ endef
 
 $(foreach f,$(asm-defines-files),$(eval $(call gen-asm-defines-file,$(f),$(out-dir)/$(sm)/include/generated/$(basename $(notdir $(f))).h)))
 
+# Device tree source file compilation
+DTC := dtc
+DTC_FLAGS += -I dts -O dtb
+DTC_FLAGS += -Wno-unit_address_vs_reg
+
+define gen-dtb-file
+# dts file path/name in $1
+# dtb file path/name in $2
+
+dtb-predts-$2 	:= $$(dir $2)$$(notdir $2).pre.dts
+dtb-predep-$2 	:= $$(dir $2)$$(notdir $2).o.d
+dtb-dep-$2	:= $$(dir $2)$$(notdir $2).d
+dtb-cmd-file-$2	:= $$(dir $2).$$(notdir $2).cmd
```

Comment:
Wrong file names, missing leading dots. Please replace these four lines with the following:
```make
dtb-basename-$2	:= $$(basename $$(notdir $2))
dtb-predts-$2   := $$(dir $2)$$(dtb-basename-$2).pre.dts
dtb-predep-$2	:= $$(dir $2).$$(dtb-basename-$2).pre.dts.d
dtb-dep-$2	:= $$(dir $2).$$(notdir $2).d
dtb-cmd-file-$2	:= $$(dir $2).$$(notdir $2).cmd
```

Reply:
you're right. thanks.